bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -385,6 +397,79 @@ impl Client {
     }
 }
 
+pub struct CreateRoomParameters {
+    pub name: String,
+    pub topic: Option<String>,
+    pub is_direct: bool,
+    pub visibility: RoomVisibility,
+    pub preset: RoomPreset,
+    pub invite: Option<Vec<String>>,
+    pub avatar: Option<String>,
+}
+
+impl From<CreateRoomParameters> for create_room::v3::Request {
```

Comment:
since this contains code that might fail  instead of using From, you could use TryFrom and implement tryFrom function.
It's the same trait but it returns a result, so you can use the commodity of the ? operator when you use the parse function.
https://doc.rust-lang.org/stable/std/convert/trait.TryFrom.html

Reply:
You mean because of the `unwrap()` I suggested? That's not ever going to fail, and https://github.com/ruma/ruma/pull/1508 introduces an infallible helper for the initial event construction that we can use here (although it also just unwraps internally).

Reply:
because of this:
```
match UserId::parse(user_id) {
                    Ok(id) => Some(id),
                    Err(e) => {
                        error!(user_id, "Skipping invalid user ID, error: {e}");
                        None
                    }
                })
                .collect(),
```

Reply:
oh yeah my bad! I forgot we want to skip this, when it fails.